src/converse-register.js

```diff
@@ -219,20 +224,37 @@
                      * Parameters:
                      *      (Submit Event) ev - Form submission event.
                      */
-                    if (ev && ev.preventDefault) { ev.preventDefault(); }
-                    var $form = $(ev.target),
-                        $domain_input = $form.find('input[name=domain]'),
-                        domain = $domain_input.val();
-                    if (!domain) {
-                        $domain_input.addClass('error');
-                        return;
+                    var cancel_label = '';
+                    if (!_converse.registration_domain) { 
+                        if (ev && ev.preventDefault) { ev.preventDefault(); }
+                        var $form = $(ev.target),
+                            $domain_input = $form.find('input[name=domain]'),
+                            domain = $domain_input.val();
+                        if (!domain) {
+                            $domain_input.addClass('error');
+                            return;
+                        }
+                        cancel_label = __('Cancel');
+                    }
+                    else {
+                        domain = _converse.registration_domain;
+                    }
```

Comment:
There's a bit more refactoring that can be done here. Lines 228 to 238 can go into a separate method (we can again call it `onProviderChosen`) which is the event handler for the `submit` event. That method can then assume that `ev` is always passed in.

Then, when you call `chooseProvider`, call it always with the domain. So in `onProviderChosen` you call `chooseProvider` with the domain that you get from the form element.

 And when you call `chooseProvider` in the case where a default domain is specified, you call `chooseProvider(_converse.registration_domain)`